Framework/API/src/ISpectrum.cpp

```diff
@@ -135,6 +135,22 @@ bool ISpectrum::hasDx() const { return bool(histogramRef().sharedDx()); }
  */
 void ISpectrum::resetHasDx() { mutableHistogramRef().setSharedDx(nullptr); }
 
+/**
+ * Makes sure the histogram has valid Y data.
+ * @param histogram A histogram to process.
+ * @throw std::invalid_argument if Y data is NULL.
+ */
+void ISpectrum::checkAndSanitizeHistogram(HistogramData::Histogram &histogram) {
```

Comment:
Ok in that case I agree we should move this back to `Histogram1D` - sorry @soininen